### PR TITLE
Add friends of friends api and runtime functions

### PR DESCRIPTION
**Pull request by sesposito**
nakama-common PR: https://github.com/heroiclabs/nakama-common/pull/148

### CODE REVIEW

File: server/api_friend.go [hunk 1]
```diff
@@ -89,6 +89,61 @@ func (s *ApiServer) ListFriends(ctx context.Context, in *api.ListFriendsRequest)
...
+		err := traceApiBefore(ctx, s.logger, s.metrics, ctx.Value(ctxFullMethodKey{}).(string), beforeFn)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	limit := 10
+	if in.GetLimit() != nil {
+		if in.GetLimit().Value < 1 || in.GetLimit().Value > 100 {
+			return nil, status.Error(codes.InvalidArgument, "Invalid limit - limit must be between 1 and 1000.")
```
Comment (syhpoon) [inline server/api_friend.go:121]: is it 100 or 1000? :thinking:

**Review by syhpoon — APPROVED** [review]
I know it's a tight timeline, but do you think there's a chance you could also squeeze in at least some basic tests of this new functionality?
If not, we can do this afterwards.

File: server/api_friend.go (hunk 1 above)
Comment (zyro) [reply server/api_friend.go:121]: For direct client operations it should be 100 maximum. Runtime calls I can see a case for allowing 1000 or even more as maximum.

**Review by zyro — APPROVED** [review]

File: server/api_friend.go (hunk 1 above)
Comment (syhpoon) [reply server/api_friend.go:121]: my point is that the returned error is misleading (containing 1000)

Comment (sesposito) [reply server/api_friend.go]: I've addressed the comment and increased the limit to 1000 for calls from the runtimes.

**Pull request merged**